assets/js/components/Select/Select.jsx

```diff
@@ -49,15 +57,21 @@ function Select({
             leaveTo="opacity-0"
           >
             <Listbox.Options className="absolute w-full py-1 mt-1 overflow-auto text-base bg-white rounded-md shadow-lg max-h-60 ring-1 ring-black ring-opacity-5 focus:outline-none sm:text-sm z-[1]">
-              {options.map((option) => (
+              {enrichedOptions.map((option) => (
                 <Listbox.Option
-                  key={option}
-                  className={({ active }) =>
-                    `cursor-default select-none relative py-2 pl-10 pr-4 ${
-                      active ? 'text-green-900 bg-green-100' : 'text-gray-900'
-                    }`
+                  key={option.value}
+                  className={({ active, disabled: optionDisabled }) =>
+                    classNames(
+                      'cursor-default select-none relative py-2 px-3',
+                      {
+                        'text-gray-400': optionDisabled,
+                        'text-green-900 bg-green-100': active,
+                        'text-gray-900': !optionDisabled && !active,
+                      }
+                    )
                   }
-                  value={option}
+                  value={option.value}
+                  disabled={!option.enabled}
```

Comment:
Usually `disabled` things are just `disabled` because it's closer to HTML native elements.

Hence:

```suggestion
                  disabled={option.disabled}
```